src/main/java/lumien/randomthings/Handler/Bloodmoon/ServerBloodmoonHandler.java

```diff
@@ -53,7 +53,8 @@ public void endWorldTick(World world) {
 
             if (bloodMoon) {
                 boolean spawnHostiles = ((WorldAccessor) world).isSpawnHostileMobs();
-                if (!Settings.BLOODMOON_RESPECT_GAMERULE || spawnHostiles) {
+                boolean doMobSpawning = world.getGameRules().getGameRuleBooleanValue("doMobSpawning");
+                if (!Settings.BLOODMOON_RESPECT_GAMERULE || spawnHostiles && doMobSpawning) {
```

Comment:
Always good to make it clear that this is the expected logic
```suggestion
                if (!Settings.BLOODMOON_RESPECT_GAMERULE || (spawnHostiles && doMobSpawning)) {
```